tests/contrib/operators/databricks_operator.py

```diff
@@ -0,0 +1,263 @@
+                                         new_cluster=NEW_CLUSTER,
+                                         notebook_task=NOTEBOOK_TASK)
+        self._test_init(op,
+                        expected_spark_jar_task=None,
+                        expected_notebook_task=NOTEBOOK_TASK,
+                        expected_existing_cluster_id=None,
+                        expected_libraries=[],
+                        expected_run_name=TASK_ID,
+                        expected_timeout_seconds=0,
+                        expected_extra_api_parameters={},
+                        expected_databricks_conn_id=DEFAULT_CONN_ID)
+
+    def test_init_with_positional_parameters(self):
+        """
+        Test the initializer which uses positional parameters. Users should not
+        use it this way but we cannot enforce this.
+        """
+        op = DatabricksSubmitRunOperator(None,
+                                         NOTEBOOK_TASK,
+                                         NEW_CLUSTER,
+                                         None,
+                                         [],
+                                         None,
+                                         0,
+                                         {},
+                                         'databricks_default',
+                                         task_id=TASK_ID)
+        self._test_init(op,
+                        expected_spark_jar_task=None,
+                        expected_notebook_task=NOTEBOOK_TASK,
+                        expected_existing_cluster_id=None,
+                        expected_libraries=[],
+                        expected_run_name=TASK_ID,
+                        expected_timeout_seconds=0,
+                        expected_extra_api_parameters={},
+                        expected_databricks_conn_id=DEFAULT_CONN_ID)
+
+    def test_init_with_invalid_oneof_task(self):
+        """
+        Test the initializer where notebook_task and spark_jar_task are
+        both specified. This is not allowed and an exception should
+        be raised.
+        """
+        run = {
+          'new_cluster': NEW_CLUSTER,
+          'notebook_task': NOTEBOOK_TASK,
+          'spark_jar_task': SPARK_JAR_TASK
+        }
+        with self.assertRaises(AirflowException):
+            DatabricksSubmitRunOperator(task_id=TASK_ID, **run)
+
+    def test_init_with_invalid_oneof_cluster(self):
+        """
+        Test the initializer where notebook_task and spark_jar_task are
+        both specified. This is not allowed and an exception should
+        be raised.
+        """
+        run = {
+          'new_cluster': NEW_CLUSTER,
+          'notebook_task': NOTEBOOK_TASK,
+          'existing_cluster_id': EXISTING_CLUSTER_ID
+        }
+        with self.assertRaises(AirflowException):
+            DatabricksSubmitRunOperator(task_id=TASK_ID, **run)
+
+    @mock.patch('airflow.contrib.operators.databricks_operator.DatabricksHook')
```

Comment:
It makes it so when airflow.contrib.operators.databricks_operator tries to look up the DatabricksHook name it will get a mock instead of the real thing. The mock has a handle as `db_mock_class` so you can modify it's behavior in test.